Add Log4j masking as a log masking approach (#3500)
wso2/docs-is: en/docs/deploy/monitor/log4j-mask-sensitive-information-in-logs.md

# Configure Log Masking with Log4j

> There can be business sensitive information that are added to logs in 

Reviewer: ```suggestion
There can be business-sensitive information that are added to logs in 
```

> the product console and/or Carbon log files. When these logs

Reviewer: ```suggestion
the product's startup logs and/or Carbon log files. When these logs
```

are analyzed, the information is exposed to those who check this.

> To avoid this potential security pitfall, users can mask sensitive

Reviewer: ```suggestion
```

> information.

Reviewer: ```suggestion
```

To avoid this potential security pitfall, users can mask sensitive

> information from the log file at the time of logging. In this feature,

Reviewer: ```suggestion
information from the log file at the time of logging.
```

> you can define patterns that need to be masked from the logs. This is

Reviewer: ```suggestion
You can define patterns that need to be masked from the logs. This is
```

particularly useful in the cases of credit card numbers, access tokens,
etc.

To configure this feature, follow the instructions given below.

---

## Enable log masking

> 1.  Log masking is enabled by default. However, you need to configure required masking patterns.

Reviewer: ```suggestion
1.  Log masking is enabled by default. However, you need to configure the required masking patterns.
```

> 2.  These patterns are configured in the `<IS-HOME>/repository/conf/wso2-log-masking.properties`. You can change 

Reviewer: `wso2-log-masking.properties` is the internal file afaiu? Users won't have to access this for any reason, right? If so, we don't need to mention this in the doc.

Author: Users might want to get an insight on what patterns to add apart from the defaults.

>     its default configurations in `<IS-HOME>/repository/conf/deployment.toml`.

Reviewer: What are these default configurations we are referring to? Is it simply that log masking is enabled or is there a default log pattern that we should know of? If there is, let's add the default configurations here.

Author: Log masking is enabled by default and users can change the default patterns in the deployment.toml.
